File: sql/mysql/Oracle/Go/SqlMode.go
package parser

// SqlMode represents SQL modes that control parsing behavior.

type SqlMode int

// Enum values for SqlMode
const (
	NoMode SqlMode = iota
	AnsiQuotes
	HighNotPrecedence
	PipesAsConcat
	IgnoreSpace
	NoBackslashEscapes
)

// String provides string representation for SqlMode
func (sm SqlMode) String() string {
	return [...]string{
		"NoMode",
		"AnsiQuotes",
		"HighNotPrecedence",
		"PipesAsConcat",
		"IgnoreSpace",
		"NoBackslashEscapes",
	}[sm]
}

File: sql/mysql/Oracle/TypeScript/SqlMode.ts
/*
 * Copyright © 2025, Oracle and/or its affiliates
 */

/* eslint-disable no-underscore-dangle */
/* cspell: ignore antlr, longlong, ULONGLONG, MAXDB */

/** SQL modes that control parsing behavior. */
enum SqlMode {
	NoMode,
	AnsiQuotes,
	HighNotPrecedence,
	PipesAsConcat,
	IgnoreSpace,
	NoBackslashEscapes
}

export default SqlMode;
